userspace/libsinsp/procfs_utils.cpp

```diff
@@ -63,3 +65,60 @@ std::string libsinsp::procfs_utils::get_systemd_cgroup(std::istream& cgroups)
 
 	return "";
 }
+
+//
+// ns_helper
+//
+libsinsp::procfs_utils::ns_helper::ns_helper(const std::string& host_root):
+	m_host_root(host_root)
+{
+	// (try to) init m_host_init_ns_mnt
+	char buf[NS_MNT_SIZE] = {0};
+	if(-1 == readlink((m_host_root + "/proc/1/ns/mnt").c_str(), buf, NS_MNT_SIZE))
+	{
+		g_logger.format(sinsp_logger::SEV_WARNING,
+				"Cannot read host init ns/mnt");
+		m_cannot_read_host_init_ns_mnt = true;
+	}
+	else
+	{
+		auto size = strlen(buf) + 1;
+		m_host_init_ns_mnt = (char*)malloc(size);
+		strncpy(m_host_init_ns_mnt, buf, size);
+	}
+}
+
+libsinsp::procfs_utils::ns_helper::~ns_helper()
+{
+	if(m_host_init_ns_mnt)
+	{
+		free(m_host_init_ns_mnt);
+		m_host_init_ns_mnt = nullptr;
+	}
+}
+
+bool libsinsp::procfs_utils::ns_helper::in_own_ns_mnt(int64_t pid) const
+{
+	if(m_host_init_ns_mnt == nullptr)
+	{
+		return false;
+	}
+
+	std::string path = m_host_root + "/proc/" + std::to_string(pid) + "/ns/mnt";
+
+	char proc_ns_mnt[NS_MNT_SIZE] = {0};
+	if(-1 == readlink(path.c_str(), proc_ns_mnt, NS_MNT_SIZE))
```

Comment:
Similar to my previous comment on `readlink`, though not as serious since it is used in `strncmp` later.
```suggestion
	if(-1 == readlink(path.c_str(), proc_ns_mnt, NS_MNT_SIZE - 1))
```